src/snakeoil/osutils/__init__.py

```diff
@@ -339,17 +339,16 @@ def unlink_if_exists(path):
 
 
 def sizeof_fmt(size, binary=True):
-    if binary:
-        units = ('B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB')
-        increment = 1024.0
-    else:
-        units = ('B', 'kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB')
-        increment = 1000.0
+    suffixes = ("k", "M", "G", "T", "P", "E", "Z", "Y")
+    increment = 1024.0 if binary else 1000.0
 
-    for i, x in enumerate(units):
-        if size < increment or i == len(units):
-            return "%3.1f %s" % (size, x)
+    suffix = ""
+    for x in suffixes:
+        if size < increment:
+            break
         size /= increment
+        suffix = f"{x.upper()}i" if binary else x
+    return "%3.1f %sB" % (size, suffix)
```

Comment:
Let's use f-strings (we support the same python versions in gentoo tree)
```suggestion
    return f"{size:3.1f} {suffix}B"
```